packages/aa-comunica-parser/lib/ComunicaSparqlParser.ts

```diff
@@ -0,0 +1,87 @@
+import type { RuleDefReturn } from '@traqula/core';
+import { createToken, GeneratorBuilder, LexerBuilder, ParserBuilder } from '@traqula/core';
+import { sparql12GeneratorBuilder } from '@traqula/generator-sparql-1-2';
+import { sparql12ParserBuilder } from '@traqula/parser-sparql-1-2';
+import { gram as g11 } from '@traqula/rules-sparql-1-1';
+import type * as T11 from '@traqula/rules-sparql-1-1';
+import { lex as l12, completeParseContext } from '@traqula/rules-sparql-1-2';
+import type { SparqlQuery } from '@traqula/rules-sparql-1-2';
+import type * as T12 from '@traqula/rules-sparql-1-2';
+import type { PatternLateral } from './ComunicaSparqlAst';
+
+// eslint-disable-next-line require-unicode-regexp
+const lateral = createToken({ name: 'Lateral', pattern: /lateral/i, label: 'Lateral pattern' });
+const lateralLexer = LexerBuilder
+  .create(l12.sparql12LexerBuilder)
+  .add(lateral);
+
+const graphPatternNotTriples: T12.SparqlRule<
```

Comment:
To be honest, I don't understand what the code below (lines 18 - 52) does exactly.
I know it defines rules for extending the parser, but it's unclear to me how this code achieves that.
It's quite cryptic to me tbh.
Could some inline comments be added to explain critical parts?
Or could some parts be abstracted if that would improve readability?